std/assembly/typedarray.ts

```diff
@@ -1260,3 +1304,14 @@ function REVERSE<TArray extends ArrayBufferView, T>(array: TArray): TArray {
   }
   return array;
 }
+
+// @ts-ignore: decorator
+@inline
+function WRAP<TArray extends ArrayBufferView, T>(buffer: ArrayBuffer, byteOffset: i32 = 0, byteLength: i32 = buffer.byteLength): TArray {
+  assert(byteOffset + byteLength <= buffer.byteLength);
```

Comment:
Ideally this should throw instead of having an assertion, even if we don't support exceptions yet. The assertion also doesn't seem to deal with all sorts of negative values being provided properly.

From a quick test on node I get

* `RangeError: Invalid typed array length` if `byteOffset + byteLength > buffer.byteLength` or `byteLength < 0`
* `RangeError: Start offset -1 is outside the bounds of the buffer` if `byteOffset < 0` or `byteOffset > buffer.byteLength`
* `RangeError: byte length of Int32Array should be a multiple of 4` if `length` is omitted and `byteLength` doesn't match a multiple of the element type
* Maybe also other cases I didn't test

Wording isn't particularly important, but using a `RangeError` here seems to be worth it to prepare for future cases.

Reply:
Added checks